### ISSUE TITLE
Trainer: Change Overall Behavior

### ISSUE THREAD

**Issue by haroldrubio** [haroldrubio/PytorchTrainer#11]
Currently, the pre-requisite is that a currently instantiated model is passed into the trainer, we require the user to pass in the name of an optimizer and the name of a loss function.

Look to re-work the behavior modeling after the recently added `set_scheduler` function. Accept a generic class name and generic list of parameters - use internal mechanisms to facilitate hyperparameter optimization of both architectures and optimizers

**Comment by haroldrubio** [comment]
This has been completed for the criterion and optimizer functions
